### PR TITLE
Add Bitcoin Output descriptor support

### PR DESCRIPTION
**Pull request by Kukks**

### CODE REVIEW

**Comment by NicolasDorier** [comment]
Concept ACK.

**Comment by Kukks** [comment]
Added tests and ready for review!

**Comment by NicolasDorier** [comment]
I thought output descriptors had a checksum. I suppose we support it though.

File: BTCPayServer/DerivationSchemeSettings.cs [hunk 1]
```diff
@@ -18,8 +18,15 @@ public static DerivationSchemeSettings Parse(string derivationStrategy, BTCPayNe
                 throw new ArgumentNullException(nameof(network));
             if (derivationStrategy == null)
                 throw new ArgumentNullException(nameof(derivationStrategy));
-            var result = network.NBXplorerNetwork.DerivationStrategyFactory.Parse(derivationStrategy);
-            return new DerivationSchemeSettings(result, network) { AccountOriginal = derivationStrategy.Trim() };
+            var result = new DerivationSchemeSettings();
+            var parser = new DerivationSchemeParser(network);
+            if (TryParseXpub(derivationStrategy, parser, ref result, false) || TryParseXpub(derivationStrategy, parser, ref result, false))
```
Comment (NicolasDorier) [inline BTCPayServer/DerivationSchemeSettings.cs:23]: `if (blah || blah)` ?

File: BTCPayServer/DerivationSchemeSettings.cs [hunk 2]
```diff
@@ -18,8 +18,15 @@ public static DerivationSchemeSettings Parse(string derivationStrategy, BTCPayNe
...
+            {
+                return result;
+            }
+
+            throw new FormatException();
```
Comment (NicolasDorier) [inline BTCPayServer/DerivationSchemeSettings.cs:28]: put descriptive message

File: BTCPayServer.Tests/AltcoinTests/AltcoinTests.cs [hunk 3]
```diff
@@ -1040,6 +1041,99 @@ public void CanParseDerivationScheme()
             Assert.Equal(
                 "tpubDDdeNbNDRgqestPX5XEJM8ELAq6eR5cne5RPbBHHvWSSiLHNHehsrn1kGCijMnHFSsFFQMqHcdMfGzDL3pWHRasPMhcGRqZ4tFankQ3i4ok-[legacy]",
                 parsed.ToString());
+            
+            //let's test output descriptor parsing support
+
+            
+            //we don't support every descriptor, only the ones which represent an HD wallet with stndard derivation paths
+            Assert.Throws<FormatException>(() => mainnetParser.ParseOutputDescriptor("pk(0279be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798)"));
```
Comment (NicolasDorier) [inline BTCPayServer.Tests/AltcoinTests/AltcoinTests.cs:1049]: is there any reason to not support it?

File: BTCPayServer/DerivationSchemeSettings.cs [hunk 4]
```diff
@@ -18,8 +18,15 @@ public static DerivationSchemeSettings Parse(string derivationStrategy, BTCPayNe
                 throw new ArgumentNullException(nameof(network));
             if (derivationStrategy == null)
                 throw new ArgumentNullException(nameof(derivationStrategy));
-            var result = network.NBXplorerNetwork.DerivationStrategyFactory.Parse(derivationStrategy);
-            return new DerivationSchemeSettings(result, network) { AccountOriginal = derivationStrategy.Trim() };
+            var result = new DerivationSchemeSettings();
```
Comment (NicolasDorier) [inline BTCPayServer/DerivationSchemeSettings.cs:21]: any reason to not simply use standard pattern with `out var result` instead?

**Comment by NicolasDorier** [comment]
test fail

File: BTCPayServer/DerivationSchemeSettings.cs (hunk 4 above)
Comment (Kukks) [reply BTCPayServer/DerivationSchemeSettings.cs:21]: It is an existing method, didn't want to refactor all usage of it

**Comment by Aquiles00770** [comment]
quiero pagar pero no se como

**Comment by Aquiles00770** [comment]
es muy complicado, si me ayudas porfa

**Pull request merged**